ppa/archive/management/commands/generate_textcorpus.py

```diff
@@ -2,199 +2,238 @@
+        parser.add_argument(
+            "--batch-size",
+            type=int,
+            default=DEFAULT_BATCH_SIZE,
+            help="Number of docs to query from solr at one time",
+        )
 
-    def save(self):
-        """Save the generated corpus text and metadata to files on disk"""
+        # add --dry-run argument (don't save anything, just iterate)
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            help="Do not save anything, just iterate over solr",
+        )
 
-        # save docs and gather metadata
-        metadata=[]
-        pdesc='Saved text to'
-        pbar=tqdm(total=self.doc_count, desc=f'{pdesc}: ...')
-        for doc_id in self.doc_ids:
-            # get saved filename and found metadata for this document
-            fn,meta = self._save_doc(doc_id)
+    #### SOLR ####
 
-            # if we saved, update progress bar desc
-            if fn: pbar.set_description(f'{pdesc}: {fn}')
+    @cached_property
+    def query_set(self):
+        """
+        A cached representation of the query set;
+        it discourages opening multiple connections to database.
+        """
+        return SolrQuerySet()
 
-            # tick
-            pbar.update()
+    def iter_solr(self, item_type="page"):
+        """
+        Iterate over solr documents of a certain `item_type`.
+        """
 
-            # add this doc's meta to metadata
-            metadata.append(meta)
-        pbar.close()
+        # get params
+        batch_size, lim = self.batch_size, self.doclimit
 
-        # save metadata csv
-        with open(self.path_metadata,'w') as of:
-            json.dump(metadata, of, indent=2)
-        print(f'Saved metadata to: {self.path_metadata}')
+        # search for this item type, order by id
+        qset = self.query_set.search(item_type=item_type).order_by("id")
 
+        # if we're looking for pages, rename those fields
+        if item_type == "page":
+            qset = qset.only(*self.PAGE_FIELDLIST)
+        elif item_type == "work":
+            qset = qset.only(*self.WORK_FIELDLIST)
 
+        # get the total count for this query
+        total = qset.count()
 
+        # if we want fewer than that, decrease "total"
+        # (this has the effect of pulling from solr only what we need,
+        # since we limit how many rows we pull by this amount)
+        if lim and total > lim:
+            total = lim
 
+        # if total smaller than batch size, decrease batch size
+        if batch_size > total:
+            batch_size = total
 
-@contextmanager
-def logging_disabled(highest_level=logging.CRITICAL):
-    """Quick way to suppress solr logs as we iterate. Taken from https://gist.github.com/simon-weber/7853144"""
-    previous_level = logging.root.manager.disable
-    logging.disable(highest_level)
-    try: yield
-    finally: logging.disable(previous_level)
+        # define a generator to iterate solr with
+        iterr = (
```

Comment:
can we give this a more semantic name? would `chunks` or `chunk_iterator` make sense here?